oci/layout/oci_src_test.go

```diff
@@ -141,3 +142,40 @@ func createImageSource(t *testing.T, sys *types.SystemContext) types.ImageSource
 	require.NoError(t, err)
 	return imageSource
 }
+
+func TestGetLocalBlobPath(t *testing.T) {
+	tmpDir := loadFixture(t, "delete_image_multiple_images")
+	ref, err := NewReference(tmpDir, "latest")
+	require.NoError(t, err)
+
+	src, err := ref.NewImageSource(context.Background(), &types.SystemContext{})
+	require.NoError(t, err)
+	defer src.Close()
+
+	// success cases
+	for _, dig := range []digest.Digest{
+		"sha256:a2f798327b3f25e3eff54badcb769953de235e62e3e32051d57a5e66246de4a1",
+		"sha256:557ac7d133b7770216a8101268640edf4e88beab1b4e1e1bfc9b1891a1cab861",
+	} {
+		path, err := GetLocalBlobPath(context.Background(), src, dig)
+		require.NoError(t, err)
+		algo, hash, _ := strings.Cut(string(dig), ":")
+		expect := filepath.Join(tmpDir, "blobs", algo, hash)
+		assert.Equal(t, expect, path)
+	}
+
+	// error cases
+	for _, dig := range []digest.Digest{
+		"sha256:as",
+		"sha256:abababababababababababababababababababababababababababababababab",
+		"sha512:a2f798327b3f25e3eff54badcb769953de235e62e3e32051d57a5e66246de4a1557ac7d133b7770216a8101268640edf4e88beab1b4e1e1bfc9b1891a1cab861",
+	} {
+		// invalid digest must error
```

Comment:
I think this comment would be better placed on the `…:as"` line.

It also might be worth commenting that the …`abab` line, and the sha512 line, is testing a valid reference to a missing blob.